app/api/infrastructure/tests/test_emis_client.py

```diff
@@ -0,0 +1,184 @@
+from json import load
+from os import environ
+from pathlib import Path
+from typing import Any
+from unittest.mock import MagicMock, patch
+
+import pytest
+from pydantic import ValidationError
+from requests import HTTPError
+
+from app.api.domain.forward_request_model import ForwardRequest
+from app.api.domain.forward_response_model import Demographics, ForwardResponse
+from app.api.infrastructure.emis_client import EmisClient
+
+
+@pytest.fixture(name="client")
+def setup_client() -> EmisClient:
+    request = ForwardRequest(
+        application_id="some application id",
+        forward_to="https://somewhere",
+        patient_nhs_number="some patient nhs number",
+        patient_ods_code="some patient ods code",
+        proxy_nhs_number="some proxy nhs number",
+    )
+    return EmisClient(request)
+
+
+def test_emis_client_get_headers(client: EmisClient) -> None:
+    """Test the EmisClient get_headers function."""
+    # Act
+    actual_result = client.get_headers()
+
+    # Assert
+    assert actual_result == {
+        "X-API-ApplicationId": "some application id",
+        "X-API-Version": "1",
+    }
+
+
+def test_emis_client_get_data(client: EmisClient) -> None:
+    """Test the EmisClient get_data function."""
+    # Act
+    actual_result = client.get_data()
+
+    # Assert
+    assert actual_result == {
+        "PatientIdentifier": {
+            "IdentifierValue": "some patient nhs number",
+            "IdentifierType": "NhsNumber",
+        },
+        "CarerIdentifier": {
+            "IdentifierValue": "some proxy nhs number",
+            "IdentifierType": "NhsNumber",
+        },
+        "PatientNationalPracticeCode": "some patient ods code",
+    }
+
+
+@patch.dict(environ, {"USE_MOCK": "True"})
+def test_emis_forward_request_use_mock(client: EmisClient) -> None:
+    """Test the EmisClient forward_request function when mock is turned on."""
+    # Arrange
+    with Path("app/api/infrastructure/data/mocked_emis_response.json").open("r") as f:
+        expected_response = load(f)
+    # Act
+    actual_result = client.forward_request()
+
+    # Assert
+    assert actual_result == expected_response
+
+
+@patch.dict(environ, {"USE_MOCK": "False"})
+@patch("app.api.infrastructure.emis_client.requests")
+def test_emis_forward_request_use_mock_on(
```

Comment:
Since mock is turned off should the name be test_emis_forward_request_use_mock_off?